src/EFCore.PG/Metadata/Conventions/NpgsqlJsonElementHackConvention.cs

```diff
@@ -20,10 +20,11 @@ public class NpgsqlJsonElementHackConvention : IPropertyAddedConvention
     public void ProcessPropertyAdded(IConventionPropertyBuilder propertyBuilder, IConventionContext<IConventionPropertyBuilder> context)
     {
         var property = propertyBuilder.Metadata;
+        var clrType = property.ClrType;
 
-        if (property.ClrType == typeof(JsonElement) && property.GetColumnType() is null)
+        if ((clrType == typeof(JsonElement) || clrType == typeof(JsonElement?)) && property.GetColumnType() is null)
         {
-            property.SetTypeMapping(_jsonTypeMapping ??= new NpgsqlJsonTypeMapping("jsonb", typeof(JsonElement)));
+            property.SetTypeMapping(_jsonTypeMapping ??= new NpgsqlJsonTypeMapping("jsonb", clrType));
```

Comment:
maybe we should keep two separate `_jsonTypeMapping` fields (one for `JsonElement` and one for `JsonElement?`)
(I'm no expert on this, but the workaround from https://github.com/npgsql/efcore.pg/issues/2977#issuecomment-1838951055 would do that)

Reply:
Yeah no I realised I got that wrong